app/controllers/course/statistics/aggregate_controller.rb

```diff
@@ -28,33 +30,65 @@ def all_students
   end
 
   def all_assessments
-    @assessments = current_course.assessments.published.includes(tab: :category)
     @all_students = current_course.course_users.students
     @all_submissions_info = all_submissions_info
 
     fetch_all_assessment_related_statistics_hash
   end
 
+  def submission_due
```

Comment:
I'm thinking that `submission_due` is also correct since we only expect 1 submission per assessment.
Nevertheless, I can change this into what you suggested as well